### PR TITLE
Multiple Event Handler [13690]

### PR DESCRIPTION
**Pull request by jparisu**

### CODE REVIEW

**Comment by jparisu** [comment]
Code Documentation and Documentation missing.

File: test/system/executable/CMakeLists.txt [hunk 1]
```diff
@@ -26,6 +26,12 @@ if(PYTHONINTERP_FOUND)
         simple_configuration
     )
 
+    # Whether one or other signal in test
+    set(SIGNAL_CONFIGURATIONS
+        "--sigterm"
+        ""
+    )
```
Comment (rsanchez15) [inline test/system/executable/CMakeLists.txt:33]: I'll create two lists here, one for the argument and another for the test name, and try to iterate both at the same time. Check [this](https://stackoverflow.com/questions/7932205/parallel-iteration-over-lists-in-makefile-or-cmake-file) to do so.

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

File: test/system/executable/launcher.ps1 [hunk 2]
```diff
@@ -22,6 +22,12 @@ Param(
     [String]
     # test_config_file
     $test_config_file
```
Comment (rsanchez15) [inline test/system/executable/launcher.ps1:24]:
```suggestion
    $test_config_file,
```

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

File: include/ddsrouter/event/MultipleEventHandler.hpp [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+     */
+    template <typename T, typename ... Args>
+    void register_event_handler(std::unique_ptr<T> handler) noexcept;
+
+protected:
+
+    /**
+     * @brief List of EventHandlers that are registered inside this.
+     *
+     * Each of this handlers belong to this object and cannot and wont be used from outside.
```
[the remark this answers is not in the document]
Comment (jparisu) [reply include/ddsrouter/event/MultipleEventHandler.hpp:96]: Do you think it is a good idea to include "strange characters" only to be grammatically correct?

[the remark this answers is not in the document]
Comment (rsanchez15) [reply include/ddsrouter/event/MultipleEventHandler.hpp:96]: Use will not.

File: test/system/executable/CMakeLists.txt (hunk 1 above)
Comment (jparisu) [reply test/system/executable/CMakeLists.txt:33]: That is what I do but with two foreach. I do not understand what do you want

File: test/system/executable/tests.py [hunk 4]
```diff
@@ -103,10 +115,17 @@ def parse_options():
         action='store_true',
         help='Print test debugging info.'
     )
+    parser.add_argument(
+        '-s',
+        '--signal',
+        type=str,
+        required=True,
+        help='Use SIGINT or SIGTERM to kill process.'
```
Comment (rsanchez15) [inline test/system/executable/tests.py:123]: SIGINT and SIGTERM (in capital letters) are not valid inputs

File: test/system/executable/tests.py [hunk 5]
```diff
@@ -103,10 +115,17 @@ def parse_options():
         action='store_true',
         help='Print test debugging info.'
     )
+    parser.add_argument(
+        '-s',
+        '--signal',
+        type=str,
```
Comment (rsanchez15) [inline test/system/executable/tests.py:121]: Instead checking if the input is string type, I'd use a function here that checks if the input is `sigint` or `sigterm` in lowercase and uppercase. 

```suggestion
        type=check_terminate_signal,
```

File: test/system/executable/tests.py [hunk 6]
```diff
@@ -229,4 +259,17 @@ def test_ddsrouter_closure(ddsrouter, configuration_file):
...
 
-    sys.exit(test_ddsrouter_closure(args.exe, args.config_file))
+    if args.signal is None:
+        logger.error(
+            'Must specify signal to kill process.')
+        sys.exit(1)
+
+    use_sigterm = False
+    if "sigterm" in args.signal:
+        use_sigterm = True
```
Comment (rsanchez15) [inline test/system/executable/tests.py:269]: Move this logic to the `check_terminate_signal()` function and don't forget to check for uppercase inputs also.

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

**Review by rsanchez15 — APPROVED** [review]
LGTM

**Pull request merged**